File: Week 2/algo-caesar-cipher/js/caesarCipher.js
const findShiftedChar = (char, shiftAmount) => {
    const charValue = char.charCodeAt() - 96
    let newValue = (charValue + shiftAmount) % 26

    if (newValue === 0) {
        newValue = 26
    } else if (newValue < 0) {
        newValue += 26}
        
    return String.fromCharCode(newValue + 96)
}

exports.caesarCipher = function(str, shiftAmount) {
    let shiftedString= ''
    for (let char of str) {
        if (char.match(/[a-z]/)) {
            const newChar = findShiftedChar(char, shiftAmount)
            shiftedString += newChar
        } else if (char.match(/[A-Z]/)) {
            shiftedString += findShiftedChar(char.toLowerCase(), shiftAmount).toUpperCase()
        } else {
            shiftedString += char
        }
    }
    return shiftedString
};

//console.log(String.fromCharCode(97)) = 'a' so 1 + 96
//console.log('a'.charCodeAt()) = 97, so -97, -96 to get 1
// console.log(exports.caesarCipher("Boy! What a string!", -5))
// console.log(findShiftedChar('b', -5))

File: Week 2/algo-caesar-cipher/python/caesar_cipher.py
def find_shifted_char(char, shift_amount):
    """
    Given a lowercase character, return the shifted character.
    """
    #ord() - 96 will give the ASCII value
    #chr() + 96 will give thhe ASCII value
    char_value = ord(char) - 96
    new_value = (char_value + shift_amount) % 26 #will wrap if need to
    if new_value == 0: #special case
        new_value = 26 #reset to the end
    return chr(new_value+96)

def caesar_cipher(string, shift_amount):
    """
    Given a string and a shift amount, return a new string with each character shifted by that amount.
    """
    shifted_string = ''

    for char in string:
        if char.islower():
            shifted_string += find_shifted_char(char, shift_amount)
        elif char.isupper():
            new_char = find_shifted_char(char.lower(), shift_amount)
            shifted_string += new_char.upper()
        else:
            shifted_string += char

    return shifted_string
    



# print(ord('b') - 96)
# print(chr(97))
# print(caesar_cipher("Boy! What a string!", -5))
# print(caesar_cipher("Hello Zach168! Yes here.", -5))
# print(find_shifted_char('e', -5))